tests/network_manager_test.go

```diff
@@ -0,0 +1,56 @@
+/*
+ *
+ * xk6-browser - a browser automation extension for k6
+ * Copyright (C) 2021 Load Impact
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package tests
+
+import (
+	"io/ioutil"
+	"testing"
+
+	"github.com/grafana/xk6-browser/testutils"
+	"github.com/grafana/xk6-browser/testutils/browsertest"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestDataURLSkipRequest(t *testing.T) {
+	t.Parallel()
+	bt := browsertest.NewBrowserTest(t)
+	p := bt.Browser.NewPage(nil)
+	t.Cleanup(func() {
+		p.Close(nil)
+		bt.Browser.Close()
+	})
+
+	logHook := &testutils.SimpleLogrusHook{HookedLevels: []logrus.Level{logrus.DebugLevel}}
+	bt.State.Logger.SetLevel(logrus.DebugLevel)
+	bt.State.Logger.AddHook(logHook)
+	bt.State.Logger.SetOutput(ioutil.Discard)
```

Comment:
We can move the logger into the `browsertest` package where it is used (without thinking about we can use it afterward (which we can but we can also think about that later when the time comes)).

And, we can do it like this instead of adding verbosity to a test (here's the [logger.patch.zip](https://github.com/grafana/xk6-browser/files/7520460/logger.patch.zip)):


```golang
func (bt *BrowserTest) Log() *DebugLog {
	logHook := &DebugLog{HookedLevels: []logrus.Level{logrus.DebugLevel}}
	bt.State.Logger.SetLevel(logrus.DebugLevel)
	bt.State.Logger.AddHook(logHook)
	bt.State.Logger.SetOutput(ioutil.Discard)
	return logHook
}
```

Within the test:

```golang
func TestDataURLSkipRequest(t *testing.T) {
	t.Parallel()
	bt := browsertest.NewBrowserTest(t)
	p := bt.Browser.NewPage(nil)
	t.Cleanup(func() {
		p.Close(nil)
		bt.Browser.Close()
	})

	log := bt.Log() // <----
	p.Goto("data:text/html,hello", nil)

	var gotMsg bool
	for _, evt := range log.Drain() {
		if evt.Message == "skipped request handling of data URL" {
			gotMsg = true
		}
	}
	assert.True(t, gotMsg)
}
```

Reply:
:-/ Renaming `SimpleLogrusHook` to `DebugLog` wouldn't make sense since the hook can be used for other log levels, and I'd like to keep the same name from k6. It's also not specific to the `browsertest` package, so it doesn't belong there.

I slightly agree that it adds a bit of setup to the test, but hooking on `DebugLevel` is unique for this test, and others might want to do it differently. The verbosity doesn't bother me, so I don't see what we gain from adding a method to `BrowserTest` :man_shrugging:

Reply:
> :-/ Renaming SimpleLogrusHook to DebugLog wouldn't make sense since the hook can be used for other log levels, 
bt.State.Logger.SetLevel(logrus.DebugLevel)
> hooking on DebugLevel is unique for this test,

Think about `DebugLevel` as a default state. Changing it in other tests won't be a problem because we can set our log levels after calling the method—We can still access `bt.State.Logger.SetLevel`.

We can call it `Log` instead of `DebugLog`. It will mean naturally like a "Browser Test's log".

> and I'd like to keep the same name from k6.

Calling it `Simple` might be OK for k6 seniors, but it doesn't tell me anything about what it represents until I read its code 🤷 

> It's also not specific to the browsertest package, so it doesn't belong there.

`browsertest` is the premiere testing environment we have right now, and all testing stuff goes right through it.

> verbosity doesn't bother me

Less code = fewer bugs 😓 I think verbosity isn't necessary here.

> I don't see what we gain from adding a method to BrowserTest

Easy to read code without thinking about what it does. Just like using functions instead of copy-pasting their inner code everywhere 😅

Reply:
OK, see the latest changes and let's compromise :stuck_out_tongue:

I'm against coupling this to the `browsertest` package, but I added a `BrowsertTest.LogHook` method, and renamed `SimpleLogrusHook` to `CacheLogrusHook`. But really man, it's a single page of code with decent documentation...

I'm not sure why this is a blocker for you. You're asking to clean up something of which there's only one use of. If the verbosity bothered someone in the future they could've done something similar then. :man_shrugging:

Reply:
Man, I'm just reviewing it. I can simply approve it. It doesn't bother me at all 😄  If you don't want me to review this deeply, just tell me, I'll act accordingly 🙇

Reply:
`CacheLogrusHook` much better by the way 😄